release_and_branching.md

```diff
@@ -0,0 +1,123 @@
+relationship between how these components are released and branched upstream.
+
+## Artifacts
+
+For each operator/repository upstream we build 3 artifict/containers:
+
+* operator container
+* bundle
+* catalog/index image (upstream only)
+
+The last artifact above (catalog/index images) are provided as a convience upstream as a means to install ad-hoc sets of 
+OpenStack services operators via projects like install_yamls. They are not used directly in the productized openstack-operator.
+The operator container and bundle are used in all deployment cases whether you install individual operators, or as a set via
+the meta openstack-operator.
+
+These containers are tagged and uploaded to quay.io under the openstack-k8s-operators organization. Each container gets tagged with the following tags:
+
+* branch name (or 'main')
+* github SHA
+* SHA256 digest of the container image itself
+
+All of the operators under openstack-k8s-operators have the above artifacts and contains. The bundle for the openstack-operator itself is a special case described below.
+
+## Bundles
+
+Each operator builds its own bundle that contains manifests including CRDs and the CSV (Cluster Service Version).
+
+All the service operators create a "normal" bundle in this fashion. The openstack-operator's bundle is a special case and does
+the following extra steps:
+
+* It contains a dependencies.yaml file which drives OLM dependencies to install all other OpenStack service operators
+* When the bundle is generated it runs a script to *merge* all the ENV variables from each service operators controller-manager. These ENV variables contain important webhook defaults that are needed within the OpenStack operator.
+* A csv-merger is used to merge in the CSV from the dataplane-operator along with all the ENV variables harvested from the above.
+
+NOTE: The dataplane-operator is merged into the CSV of the openstack-operator so that it's CRD can be treated
+as a top level visible resource in the OpenShift UI.
+
+NOTE: We no longer merge all the CSV for all openstack service operators because we hit the bundle size limit. This is a limitation due to how OLM relies on ConfigMaps when unpacking bundles. (separate bundles are required for OpenStack operators at this time)
+
+## How go.mod is used to pin the go dependencies and control the service operator containers that get installed
+
+Each service operator that is integrated into the openstack-operator has an entry in the go.mod for the openstack-operator.  We required the use of ([psuedoversions] https://go.dev/ref/mod#pseudo-versions) in the go.mod file for all openstack service operator dependencies. In addition to controlling how the golang dependency gets installed the psuedoversion controls which bundle is used and installed with the openstack-operator and acts as a means to control and promote new service operator builds to be used.
```

Comment:
The link here should use this format: `[display text](url)`

Reply:
you know, I tried creating this with vscode and see what happens. fixed, and vim it is next time